doc/company.texi

```diff
@@ -0,0 +1,1499 @@
+@end lisp
+@end defopt
+
+@defopt company-selection-wrap-around
+Enable this option to loop (cycle) the candidates' selection: after
+selecting the last candidate on the list, a command to select the next
+candidate does so with the first candidate.  By default, this option
+is disabled, which means the selection of the next candidate stops on
+the last item.  The selection of the previous candidate is influenced
+by this option similarly.
+@end defopt
+
+@defopt company-require-match
+To allow typing in characters that don't match the candidates, set the
+value of this option to @code{nil}.  For an opposite behavior (that
+is, to disallow non-matching input), set it to @code{t}.  By default,
+Company is configured to require a matching input only if a user
+manually enables completion or selects a candidate; by having the
+option configured to call the function
+@code{company-explicit-action-p}.
+@end defopt
+
+@defopt company-lighter-base
+This user options allows to configure a string indicator of the
+enabled @emph{company-mode} in the mode line.  The default value is
+@samp{company}.
+@end defopt
+
+@anchor{company-auto-commit}
+@defopt company-auto-commit
+One more pair of the user options may instruct Company to complete
+with the selected candidate by typing one of the
+@code{company-auto-commit-chars} @footnote{The options
+@code{company-auto-commit} and @code{company-auto-commit-chars} used
+to be called @code{company-auto-complete} and
+@code{company-auto-complete-chars} respectively, which was in more
+accordance with the terminology given in this manual.  But the
+resulting combination of the words @samp{auto-complete} present in
+those names made it seem the role of these user options was to
+configure Company's auto-start behavior.  Hence, it was chosen to
+rename the options to, hopefully, less confusing names.}.  The user
+option @code{company-auto-commit} can be enabled or disabled by
+setting its value to one of: @code{nil}, @code{t}, or a predicate
+function name.
+@ifnothtml
+@pxref{Wrong Type of Argument,Predicate,,eintr}.
+@end ifnothtml
+@ifhtml
+See @uref{https://www.gnu.org/software/emacs/manual/html_node/eintr/Wrong-Type-of-Argument.html,
+(eintr)Predicate}.
+@end ifhtml
+@end defopt
+
+@anchor{company-auto-commit-chars}
+@defopt company-auto-commit-chars
+This option acts only when @code{company-auto-commit} is enabled.  The
+value can be one of: a string of characters, a list of syntax
+description characters (@pxref{Syntax Class Table,,,elisp}), or a
+predicate function.  By default, @code{company-auto-commit-chars} is
+set to the list of the syntax characters: @w{@code{(?\ ?\) ?.)}},
+which translates to the whitespaces, close parenthesis, and
+punctuation.  The particular convenience of
+@code{company-auto-commit-chars} is that they do not act as a trigger
+when they are a part of valid completion.
+@end defopt
+
+@c TODO: Maybe also describe the following options.
```

Comment:
Up to you. But if you are going to do that, I'd say the first 3 are probably more important than the last two.

As in, more likely to be customized.

Reply:
> I'd say the first 3 are probably more important than the last two.

Thanks, good to know.

I'm removing this comment too (I don't like todos or fixmes hanging around in the source).